hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/RowIndexSeekerV1.java

```diff
@@ -154,19 +153,6 @@ private int binarySearch(Cell seekCell, boolean seekBefore) {
     }
   }
 
-  private int compareRows(ByteBuffer row, Cell seekCell) {
```

Comment:
I see. This method is moved to CellComparator. Is that required? This impl here correctly handle things. Within the BBUtils we are handling HBB/DBB. But in this patch, it is trying to handle in CellComparator which is not good IMHO.
Also within this context we know the passed row BB is sliced for the rk bytes alone. Within CellComparator we can not have such assumption. So its better to keep it here only.  I believe only this class is having such need for compares. Any other place?

Reply:
We can't have encoders doing their own logic figuring Cell Compare else the issue this patch fixes will perpetually haunt us as encoding implementors make the same mistake of around comparators over and over again; i.e. presuming only one comparator in the system or knowing there are two at least but defaulting to user-space comparator since that is what is used 99.9% of the time.

CellComparator already has methods that do row compare. This is just adding one that hosts the row in ByteBuffer instead of an array... an override.

CellComparators need to do more ByteBuffering, not less. CellComparator needs to learn how to do compares w/o making copies of the ByteBuffer content as it currently does in BBUtills if the BB doesn't have an array because  comparators don't know how to stride through a ByteBuffer.

bq. Also within this context we know the passed row BB is sliced for the rk bytes alone. Within CellComparator we can not have such assumption. 

CellComparator has one method that does this already taking an array.

Reply:
I understood ur arg now.  Make sense...  The MetaCellComparator was doing on heap copy. So ur impl there is ok as of now. May be we can see whether we can avoid some of those. But can be another jira.  +1 for doing this way boss.

Reply:
Thanks @anoopsjohn 

Will put up new patch that implements your suggestions and those of @HorizonNet